src/main/java/com/github/rholder/retry/Retryer.java

```diff
@@ -40,101 +39,47 @@
+     * @param rejectionPredicates the predicates used to decide if the attempt must be rejected
+     *                            or not. If an attempt is rejected, the retryer will retry the call,
+     *                            unless the stop strategy indicates otherwise or the thread is
+     *                            interrupted.
+     * @param listeners           collection of retry listeners
      */
-    public Retryer(@Nonnull AttemptTimeLimiter attemptTimeLimiter,
-                   @Nonnull StopStrategy stopStrategy,
-                   @Nonnull WaitStrategy waitStrategy,
-                   @Nonnull BlockStrategy blockStrategy,
-                   @Nonnull Predicate<Attempt<V>> rejectionPredicate) {
-        this(attemptTimeLimiter, stopStrategy, waitStrategy, blockStrategy, rejectionPredicate, new ArrayList<>());
-    }
-
-    /**
-     * Constructor
-     *
-     * @param attemptTimeLimiter to prevent from any single attempt from spinning infinitely
-     * @param stopStrategy       the strategy used to decide when the retryer must stop retrying
-     * @param waitStrategy       the strategy used to decide how much time to sleep between attempts
-     * @param blockStrategy      the strategy used to decide how to block between retry attempts; eg, Thread#sleep(), latches, etc.
-     * @param rejectionPredicate the predicate used to decide if the attempt must be rejected
-     *                           or not. If an attempt is rejected, the retryer will retry the call, unless the stop
-     *                           strategy indicates otherwise or the thread is interrupted.
-     * @param listeners          collection of retry listeners
-     */
-    @Beta
-    public Retryer(@Nonnull AttemptTimeLimiter attemptTimeLimiter,
-                   @Nonnull StopStrategy stopStrategy,
-                   @Nonnull WaitStrategy waitStrategy,
-                   @Nonnull BlockStrategy blockStrategy,
-                   @Nonnull Predicate<Attempt<V>> rejectionPredicate,
-                   @Nonnull Collection<RetryListener> listeners) {
+    Retryer(@Nonnull AttemptTimeLimiter attemptTimeLimiter,
+            @Nonnull StopStrategy stopStrategy,
+            @Nonnull WaitStrategy waitStrategy,
+            @Nonnull BlockStrategy blockStrategy,
+            @Nonnull List<Predicate<Attempt<?>>> rejectionPredicates,
+            @Nonnull Collection<RetryListener> listeners) {
         Preconditions.checkNotNull(attemptTimeLimiter, "timeLimiter may not be null");
         Preconditions.checkNotNull(stopStrategy, "stopStrategy may not be null");
         Preconditions.checkNotNull(waitStrategy, "waitStrategy may not be null");
         Preconditions.checkNotNull(blockStrategy, "blockStrategy may not be null");
-        Preconditions.checkNotNull(rejectionPredicate, "rejectionPredicate may not be null");
+        Preconditions.checkNotNull(rejectionPredicates, "rejectionPredicate may not be null");
```

Comment:
plural in the message too?

Reply:
Thanks! I had no idea anybody was paying any attention.

Reply:
Thank you for picking up this project and giving it some TLC.